homeassistant/components/bryant_evolution/climate.py

```diff
@@ -206,6 +142,7 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
             )
         self._attr_hvac_mode = hvac_mode
         self._async_write_ha_state()
+        await self.coordinator.async_request_refresh()
```

Comment:
I don't think the controls will jump back -- we're calling async_write_ha_state() after we set the attribute to the new value. They also behave correctly when I test it locally. Am I missing something?

Reply:
What I wanted to say is, you don't even have to set the value and don't need to call `async_write_ha_state`, only do a `async_refresh`. The frontend will stay at the state that was set and after the coordinator refresh it will be again in sync. (Maybe this works in combination with properties, not sure about that)

Have a look here:
https://github.com/home-assistant/core/blob/9dd13af50ca06abc540b3df1abfbee3d1d595893/homeassistant/components/pyload/switch.py#L87-L142